src/theory/datatypes/datatypes_sygus.cpp

```diff
@@ -895,8 +895,9 @@ Node SygusSymBreakNew::registerSearchValue(
               Trace("sygus-rr-debug")
                   << "; unsound: are not equivalent for : " << std::endl;
               std::vector<Node> vars;
+              its->second.getVariables(vars);
```

Comment:
It looks like `vars` is only used for the assertion check. Move it into an `if (Configuration::isAssertionBuild())`?

Reply:
It's also used in the printing within the for loop: https://github.com/CVC4/CVC4/pull/2326/files#diff-b5f4e26d011c1b54d669904177e990efL903.

Reply:
Ah yes, but then why not just move that function call into an if statement guarded by `Configuration::isAssertionBuild() || Trace.on("sygus-rr-debug")`? Like that we don't incur the overhead of copying the variables when we are running in production without tracing.

Reply:
Actually, this output should be part of the exception on line 921 (I'm surprised its not already). I'll refactor.